widget/embedded/src/components/RouteErrors/RouteErrorsModal.tsx

```diff
@@ -0,0 +1,109 @@
+import type { ModalPropTypes } from './RouteErrors.types';
+
+import { i18n } from '@lingui/core';
+import {
+  BottomLogo,
+  Button,
+  Divider,
+  MessageBox,
+  Modal,
+  Typography,
+  WarningIcon,
+} from '@rango-dev/ui';
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { navigationRoutes } from '../../constants/navigationRoutes';
+import { numberToString } from '../../utils/numbers';
+
+import { Flex } from './RouteErrors.styles';
+import { RouteErrorsModalItem } from './RouteErrorsModalItem';
+
+const WARNING_LEVEL_LIMIT = -10;
+const MAX_DECIMAL = 2;
+const MIN_DECIMAL = 0;
+
+export function RouteErrorsModal(props: ModalPropTypes) {
+  const {
+    highValueLoss,
+    open,
+    onToggle,
+    percentageChange,
+    inputUsdValue,
+    outputUsdValue,
+    totalFeeInUsd,
+  } = props;
+  const navigate = useNavigate();
+  const type =
+    highValueLoss && !!percentageChange?.lt(WARNING_LEVEL_LIMIT)
+      ? 'error'
+      : 'warning';
+  const highValueLossData = [
+    { title: 'Swapping', value: numberToString(inputUsdValue) },
+    {
+      title: 'Gas cost',
+      value: numberToString(totalFeeInUsd, MIN_DECIMAL, MAX_DECIMAL),
+    },
+    { title: 'Receiving', value: numberToString(outputUsdValue) },
+    {
+      title: 'Price impact',
+      value: numberToString(percentageChange, MIN_DECIMAL, MAX_DECIMAL),
+      valueColor: `${type}500`,
+    },
+  ];
+
+  return (
+    <Modal
+      open={open}
+      onClose={() => onToggle(false)}
+      container={document.getElementById('swap-box') || document.body}>
+      {highValueLoss ? (
+        <MessageBox
+          type={type}
+          title={i18n.t('High Price Impact')}
+          description={i18n.t(
+            `The price impact is significantly higher than the allowed amount. If you are sure, continue, otherwise, change the swap.`
```

Comment:
Please double check your PR and move all these error messages to a file similar to `error.ts`.